libs/middleware/response_date.go

```diff
@@ -0,0 +1,16 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+)
+
+// SetResponseDate sets a header date in the response
+func SetResponseDate() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Add("date", time.Now().Format(http.TimeFormat))
+			next.ServeHTTP(w, r)
+		})
+	}
+}
```

Comment:
It would have been better to allow passing in time and format so that you can adjust and test. And then, provide a helper with the desired defaults, which is a common pattern. This provides better testability and allows for customisation without having to update code.

```golang
// SetResponseDate sets a header date in the response to now formatted with http.TimeFormat.
func SetResponseDate() func(http.Handler) http.Handler {
	return SetResponseDateTimeFmt(time.Now(), http.TimeFormat)
}

// SetResponseDateTimeFmt sets a header date in the response to when formatted with tmpl.
func SetResponseDateTimeFmt(when time.Time, tmpl string) func(http.Handler) http.Handler {
	return func(next http.Handler) http.Handler {
		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			w.Header().Add("date", when.Format(tmpl))
			next.ServeHTTP(w, r)
		})
	}
}
```